src/event/event_kevent.c

```diff
@@ -1887,10 +1916,17 @@ _dispatch_event_loop_poke(dispatch_wlh_t wlh, uint64_t dq_state, uint32_t flags)
 {
 	if (wlh == DISPATCH_WLH_MANAGER) {
 		dispatch_kevent_s ke = (dispatch_kevent_s){
+#ifdef EVFILT_USER
 			.ident  = 1,
 			.filter = EVFILT_USER,
 			.fflags = NOTE_TRIGGER,
 			.udata = (dispatch_kevent_udata_t)DISPATCH_WLH_MANAGER,
+#else
+			.ident = DISPATCH_KEVENT_ERSATZ_EVFILT_USER_MASK,
```

Comment:
Because not all kqueue implementations have EVFILT_USER. We have to make do on OpenBSD by using a timer with a short timeout to fake it.

Reply:
I understand but I'm saying you don't even need kqueue for the thing that this is trying to use EVFILT_USER for. On Darwin, we decided to use the EVFILT_USER to bring up the manager thread through our kernel-integrated workqueue subsystem. 

On linux and OpenBSD, you don't need that - all the caller of this function with `DISPATCH_WLH_MANAGER` trying to do, is wake up the manager queue so let's just do that and enqueue it onto the manager root queue and have the workqueue bring up threads for it. Why are we bothering with kevent and timers and all that additional kernel overhead?

Reply:
I'm not sure I understand. A lot of Dispatch internals are undocumented and difficult to understand, especially in the event backend code like this, so I may be confused at your suggestion. 

All of the other event backends (Windows, Linux's epoll) appear to use platform-specific functionality to handle other signal inputs in the event loop (fd read/write, signals, timers), and presumably in `_dispatch_event_loop_poke` this is done with the same platform-specific mechanism so that all signals are ganged together in the one event loop. And the kevent backend that Darwin and OpenBSD use is no different.

Are you saying there is some way that _all_ of these event backends can avoid having to use the platform-specific functionality here? If so, knowing how to improve this is less than obvious and if that is the path you would want to see me go down, I would need more guidance on that. 

That being said, I could try and stare at this code a little more to try and figure out how to implement the suggestion you made, but from a perspective of someone trying to get Dispatch to build on a new platform without breaking an existing one, as well as having those changes work and be reliable, the smallest change to an existing codebase is likely the safest one.

This is why just providing a compatible alternative to EVFILT_USER was done, because it seems like the simplest, cleanest way to get this backend to work on this platform. 

(One alternative I briefly considered was to create an entirely new backend based solely on portable POSIXish primitives, but knowing how to properly implement the semantics of that backend API interface was baffling enough as it is, and ultimately it made much more sense to tweak the kevent backend to be a little more portable.)